cmd/shell.go

```diff
@@ -209,42 +209,49 @@ func FailOnError(err error, msg string) {
 func AmqpChannel(conf Config) (*amqp.Channel, error) {
 	var conn *amqp.Connection
 
-	if conf.AMQP.SSL != nil {
-		if strings.HasPrefix(conf.AMQP.Server, "amqps") == false {
-			err := fmt.Errorf("SSL configuration provided, but not using an AMQPS URL")
+	if conf.AMQP.TLS == nil {
+		// Configuration did not specify SSL options
+		conn, err := amqp.Dial(conf.AMQP.Server)
+		if err != nil {
 			return nil, err
 		}
 
-		cfg := new(tls.Config)
+		return conn.Channel()
+	}
 
-		cert, err := tls.LoadX509KeyPair(conf.AMQP.SSL.CertFile, conf.AMQP.SSL.KeyFile)
-		if err != nil {
-			err = fmt.Errorf("Could not load Client Certificates: %s", err)
-			return nil, err
-		}
-		cfg.Certificates = append(cfg.Certificates, cert)
+	// They provided SSL options, so let's load them.
+	blog.GetAuditLogger().Info("Loading SSL Options.")
+
+	if strings.HasPrefix(conf.AMQP.Server, "amqps") == false {
+		err := fmt.Errorf("SSL configuration provided, but not using an AMQPS URL")
+		return nil, err
+	}
+	if len(conf.AMQP.TLS.CertFile) == 0 || len(conf.AMQP.TLS.KeyFile) == 0 {
+		err := fmt.Errorf("Configuration values AMQP.SSL.KeyFile and AMQP.SSL.CertFile may not be nil.")
+		return nil, err
+	}
 
-		if conf.AMQP.SSL.CACertFile != nil {
-			cfg.RootCAs = x509.NewCertPool()
+	cfg := new(tls.Config)
 
-			ca, err := ioutil.ReadFile(*conf.AMQP.SSL.CACertFile)
-			if err != nil {
-				err = fmt.Errorf("Could not load CA Certificate: %s", err)
-				return nil, err
-			}
-			cfg.RootCAs.AppendCertsFromPEM(ca)
-		}
+	cert, err := tls.LoadX509KeyPair(conf.AMQP.TLS.CertFile, conf.AMQP.TLS.KeyFile)
+	if err != nil {
+		err = fmt.Errorf("Could not load Client Certificate or Key: %s", err)
+		return nil, err
+	}
+	cfg.Certificates = append(cfg.Certificates, cert)
+
+	if conf.AMQP.TLS.CACertFile != nil {
+		cfg.RootCAs = x509.NewCertPool()
```

Comment:
I'm just going to make all three pointers to strings, so they behave the same way and don't cause future confusion.